### PR TITLE
Add map annotations to Local screen

### PR DESCRIPTION
**Pull request by pdamonkey**
Annotations added to map from the local endpoint.  
Selectable with a More button to link to included URL. 

swift-leeds-map-annotations

swift-leeds-map-selection

### CODE REVIEW

File: SwiftLeeds/Views/Local/LocalView.swift [hunk 1]
```diff
@@ -7,29 +7,66 @@
...
                     categories: model.categories,
                     error: model.error,
                     maxHeight: geometry.size.height * Constants.maxHeightRatio
                 )
+
                 if model.error != nil {
                     errorView
                 }
             }
         }
```
Comment (SwiftyAlex) [inline SwiftLeeds/Views/Local/LocalView.swift:69]: If we add a cheeky animation modifier here:
```
 .animation(.easeInOut, value: selectedLocation)
```

And one of these lovely things below line 53:
```
.transition(.opacity)
```

We'll get this fading in for free 👯

File: SwiftLeeds/Views/Local/LocalViewModel.swift [hunk 2]
```diff
@@ -8,23 +8,35 @@
...
+
+            DispatchQueue.main.async {
+                self.categories = localResults.data
+                self.selectedCategory = self.categories.first
+            }
         } catch {
             self.error = error
         }
+
         await publishResults()
```
Comment (SwiftyAlex) [inline SwiftLeeds/Views/Local/LocalViewModel.swift:33]: This `publishResults` was only here because I wasn't using `Published`, leaving this in alongside `Published` will cause a third update which isn't needed

The `self.categories = localResults.data` will already cause the UI to update due to the property wrapper, and same for `self.selectedCategory = self.categories.first`

Background on this area as to why/how I did it the other way here: 

https://www.hackingwithswift.com/quick-start/swiftui/how-to-send-state-updates-manually-using-objectwillchange

TLDR `@Published` _or_ `objectWillChange.send()`, but generally not both

File: SwiftLeeds/Views/Local/LocalView.swift [hunk 3]
```diff
@@ -7,29 +7,66 @@
...
+    @State private var selectedLocation: Local.Location?
+
     var body: some View {
-        GeometryReader{ geometry in
-            ZStack {
-                Map(coordinateRegion: $mapRegion)
+        ZStack {
+            GeometryReader{ geometry in
+                if let category = model.selectedCategory {
+                    Map(coordinateRegion: $mapRegion, showsUserLocation: true, annotationItems: category.locations) { location in
```
Comment (SwiftyAlex) [inline SwiftLeeds/Views/Local/LocalView.swift:23]: A neat way to get around this if let with the same view would be to simply publish `currentLocations` from the model instead, allowing one `Map` with one array, instead of having the two.  

This would mean shifting `selectedLocation` into the model too, and then wherever you set `selectedLocation` you could just call `setLocation(location)` or use a `didSet`, or I'm sure there's a bunch of other ways to do the same thing!

File: SwiftLeeds/Views/Local/LocalView.swift [hunk 4]
```diff
@@ -7,29 +7,66 @@
...
+                        MapAnnotation(coordinate: location.location.coordinate) {
+                            Image(uiImage: UIImage(systemName: category.symbolName) ?? UIImage(imageLiteralResourceName: category.symbolName))
+                                .frame(width: 44, height: 44)
+                                .background(
+                                    RoundedRectangle(cornerRadius: 6)
+                                        .fill(.white)
+                                )
+                                .onTapGesture {
+                                    selectedLocation = location
+                                    print(mapRegion)
```
Comment (SwiftyAlex) [inline SwiftLeeds/Views/Local/LocalView.swift:33]: Cheeky print snuck in 👀

File: SwiftLeeds/Views/Local/LocalView.swift [hunk 5]
```diff
@@ -57,6 +94,34 @@ struct LocalView: View {
             await model.loadData()
         }
     }
+
+    private func locationInfoView(category: Local.LocationCategory, location: Local.Location) -> some View {
```
Comment (SwiftyAlex) [inline SwiftLeeds/Views/Local/LocalView.swift:98]: It might be nicer to have this as a full `struct LocationView` which just gets passed these params instead, that way it can live in its own file with its own preview etc?

File: SwiftLeeds/Views/Local/LocalViewModel.swift [hunk 6]
```diff
@@ -8,23 +8,35 @@
...
     }
 
+    var annotationItem: Local.Location {
```
Comment (SwiftyAlex) [inline SwiftLeeds/Views/Local/LocalViewModel.swift:36]: Is that a....force unwrap 👀

File: SwiftLeeds/Views/Local/LocalView.swift [hunk 7]
```diff
@@ -57,6 +94,34 @@ struct LocalView: View {
...
+        VStack(spacing: 10) {
+            Image(uiImage: UIImage(systemName: category.symbolName) ?? UIImage(imageLiteralResourceName: category.symbolName))
+                .renderingMode(.template)
+                .frame(width: 44, height: 44)
```
Comment (SwiftyAlex) [inline SwiftLeeds/Views/Local/LocalView.swift:102]: We'll need to have some aspect ratio stuff here - when you hard set the width and height like this, you'll be skewing(stretching? idk the word to use here) symbols that aren't squares

An easy way is below but theres many ways to achieve the same thing

```swift
.resizable()
.aspectRatio(contentMode: .fit)
.frame(maxWidth: 64, maxHeight: 64) // may have to pass alignment too depending on container
```

**Review by SwiftyAlex — COMMENTED** [review]
Just some thoughts, feel free to dismiss

File: SwiftLeeds/Views/Local/LocalViewModel.swift (hunk 6 above)
Comment (pdamonkey) [reply SwiftLeeds/Views/Local/LocalViewModel.swift:36]: That was used for initial testing of the Map, taken out

File: SwiftLeeds/Views/Local/LocalView.swift (hunk 7 above)
Comment (pdamonkey) [reply SwiftLeeds/Views/Local/LocalView.swift:102]: We're using SFSymbols, and intent was for them all to be square

Comment (pdamonkey) [reply SwiftLeeds/Views/Local/LocalView.swift:102]: All the ones we're using this year work well

File: SwiftLeeds/Views/Local/LocalView.swift (hunk 1 above)
Comment (pdamonkey) [reply SwiftLeeds/Views/Local/LocalView.swift:69]: I'm not sure which part this is referring to?

File: SwiftLeeds/Views/Local/LocalViewModel.swift (hunk 2 above)
Comment (pdamonkey) [reply SwiftLeeds/Views/Local/LocalViewModel.swift:33]: Remove redundant objectWillChange.send()

File: SwiftLeeds/Views/Local/LocalView.swift (hunk 3 above)
Comment (pdamonkey) [reply SwiftLeeds/Views/Local/LocalView.swift:23]: Refactored

File: SwiftLeeds/Views/Local/LocalView.swift (hunk 5 above)
Comment (pdamonkey) [reply SwiftLeeds/Views/Local/LocalView.swift:98]: Happy with where this is for now

**Review by adamrushy — APPROVED** [review]
LGTM ✅ but animation comment from @SwiftyAlex would be nice
